renderapi/transform.py

```diff
@@ -1370,6 +1370,127 @@ def estimate_dstpts(transformlist, src=None):
     return dstpts
 
 
+class NonLinearTransform(Transform):
+    """
+    render-python class that implements the mpicbg.trakem2.transform.nonLinearTransform class
+    
+    Parameters
+    ----------
+    dataString:str or None
+        data string of transformation
+    json:dict or NOne
+        json compatible dictionary representation of the transformation
+
+    Returns
+    -------
+    :class:`NonLinearTransform`
+        a transform instance
+
+
+    """
+
+    className = 'mpicbg.trakem2.transform.nonLinearTransform'
+
+
+
+
+    def __init__(self, dataString=None, json=None,transformId=None):
+        if json is not None:
+            self.from_dict(json)
+        else:
+            if dataString is not None:
+                self._process_dataString(dataString)
+        self.transformId = transformId
+
+    def _process_dataString(self, dataString):
+        # trailing whitespace in string.... for some reason
+        fields = dataString.split(" ")[:-1]
+        
+        self.dimension = int(fields[0])
+        self.length = int(fields[1])
+
+        # last 2 fields are width and height
+        self.width = int(fields[-2])
+        self.height = int(fields[-1])
+        
+        data = np.array(fields[2:-2],dtype='float32')
+        self.beta=data[0:2*self.length].reshape(self.length,2)
+        assert(self.beta.shape[0]==self.length)

```

Comment:
Can we have these raise RenderErrors (or a subclass of it) rather than be assertions?